sentry-ruby/lib/sentry/profiler/helpers.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require "securerandom"
+
+module Sentry
+  class Profiler
+    module Helpers
+      def in_app?(abs_path)
+        abs_path.match?(@in_app_pattern)
```

Comment:
Since this helper module relies on multiple instance variables, I feel in this case it's better to create a base profiler class that has these helpers but also define the ivars in the constructor.

Reply:
@st0012 yeah I was planning to make a bigger refactoring of these classes in a separate PR but I can introduce an abstract class already here. Definitely better than a Helper module :)

Reply:
I ended up postponing this because Vernier uses its Output class to generate the profile hash, and that one uses Helpers. What I'd like to do eventually is to refactor both profiler classes so that use the same structure (like using Output class for example) and then unifying them will become simpler.